lua/telescope/builtin/internal.lua

```diff
@@ -81,7 +81,15 @@ internal.builtin = function(opts)
     previewer = previewers.builtin.new(opts),
     sorter = conf.generic_sorter(opts),
     attach_mappings = function(_)
-      actions.select_default:replace(actions.run_builtin)
+      actions.select_default:replace(function(prompt_bufnr)
+        local selection = action_state.get_selected_entry()
+        if not selection then
+          print "[telescope] Nothing currently selected"
+          return
+        end
+        -- Call appropriate telescope builtin
+        require("telescope.builtin")[selection.text](opts)
```

Comment:
You still need this part:

```lua
  if string.match(selection.text, " : ") then
    -- Call appropriate function from extensions
    local split_string = vim.split(selection.text, " : ")
    local ext = split_string[1]
    local func = split_string[2]
    require("telescope").extensions[ext][func]()
  else
    -- Call appropriate telescope builtin
    require("telescope.builtin")[selection.text]()
  end
```
so extensions also work

Reply:
Yeah, but I figured that they don't really fit in an internal function called `builtin`. 

Maybe I'm missing something obvious here, but I can see `fzf` showing up, but then it doesn't do anything. It also doesn't work on `zoxide` until I actually run it manually first, so that makes it a bit useless for extensions discoverability since it requires that you know and execute the command first.

Do you still want me to keep the ability to run extensions? or remove their related opts from this function instead?

Reply:
`fzf` is shown because of a bad design decision in the extension interface. Pickers and internal functions will be exported the same way. But this shouldnt be addressed here. It works for me:
https://user-images.githubusercontent.com/15233006/145681549-3c991917-5e77-4708-a0d8-b6e4eba3196e.mp4

> Do you still want me to keep the ability to run extensions?

Where do i say i wanna drop this ability. Especially without adjusting documentation / removal of this option

Reply:
I'll leave it in then.